examples/src/test/scala/DoobieExample.scala

```diff
@@ -29,77 +29,86 @@ import scala.concurrent.{ExecutionContext}
-
-  val createTransactor: IO[Transactor[IO]] =
-    H2Transactor[IO]("jdbc:h2:mem:test;DB_CLOSE_DELAY=-1", "sa", "")
-
+object DatabaseExample {
   case class AuthorId(id: Int)
   case class Author(id: Int, name: String)
 
-  val dropTable = sql"DROP TABLE IF EXISTS author".update.run
-
-  val createTable = sql"""
-     CREATE TABLE author (
-       id INTEGER PRIMARY KEY,
-       name VARCHAR(20) NOT NULL UNIQUE
-     )
-    """.update.run
+  object Queries {
+    implicit val authorIdMeta: Meta[AuthorId] =
+      Meta[Int].xmap(AuthorId(_), _.id)
 
-  def addAuthor(author: Author) =
-    sql"INSERT INTO author (id, name) VALUES(${author.id}, ${author.name})".update.run
+    def fetchById(id: AuthorId): ConnectionIO[Option[Author]] =
+      sql"SELECT * FROM author WHERE id = $id".query[Author].option
 
-  val authors: List[Author] =
-    List("William Shakespeare", "Charles Dickens", "George Orwell").zipWithIndex.map {
-      case (name, id) => Author(id + 1, name)
+    def fetchByIds(ids: NonEmptyList[AuthorId]): ConnectionIO[List[Author]] = {
+      val q = fr"SELECT * FROM author WHERE" ++ Fragments.in(fr"id", ids)
+      q.query[Author].list
     }
+  }
 
-  val xa: Transactor[IO] =
-    (for {
-      xa <- createTransactor
-      _  <- (dropTable *> createTable *> authors.traverse(addAuthor)).transact(xa)
-    } yield xa).unsafeRunSync()
+  object Database {
+    val createTable = sql"""
+       CREATE TABLE author (
+         id INTEGER PRIMARY KEY,
+         name VARCHAR(20) NOT NULL UNIQUE
+       )
+      """.update.run
 
-  val authorDS = new DataSource[AuthorId, Author] {
-    override def name = "AuthorDoobie"
+    val dropTable = sql"DROP TABLE IF EXISTS author".update.run
 
-    override def fetch[F[_]: ConcurrentEffect](id: AuthorId): F[Option[Author]] =
-      LiftIO[F].liftIO(fetchById(id).transact(xa))
+    def addAuthor(author: Author) =
+      sql"INSERT INTO author (id, name) VALUES(${author.id}, ${author.name})".update.run
 
-    override def batch[F[_]: ConcurrentEffect](
-        ids: NonEmptyList[AuthorId]): F[Map[AuthorId, Author]] =
-      LiftIO[F].liftIO(
-        fetchByIds(ids)
-          .map { authors =>
-            authors.map(a => AuthorId(a.id) -> a).toMap
-          }
-          .transact(xa)
-      )
+    val authors: List[Author] =
+      List("William Shakespeare", "Charles Dickens", "George Orwell").zipWithIndex.map {
+        case (name, id) => Author(id + 1, name)
+      }
 
-    def fetchById(id: AuthorId): ConnectionIO[Option[Author]] =
-      sql"SELECT * FROM author WHERE id = $id".query[Author].option
+    def createTransactor[F[_]: ConcurrentEffect] =
+      H2Transactor[F]("jdbc:h2:mem:test;DB_CLOSE_DELAY=-1", "sa", "")
 
-    def fetchByIds(ids: NonEmptyList[AuthorId]): ConnectionIO[List[Author]] = {
-      val q = fr"SELECT * FROM author WHERE" ++ Fragments.in(fr"id", ids)
-      q.query[Author].list
-    }
+    def transactor[F[_]: ConcurrentEffect]: F[Transactor[F]] =
+      for {
```

Comment:
```suggestion
      createTransactor.flatTap { _.trans(dropTable *> createTable *> authors.traverse(addAuthor) )
```
The `flatTap` allows you to _tee_ and use the output for a side-effectful computation.

Reply:
I tried it but I got a type mismatch between `doobie.h2.H2Transactor` and `doobie.Transactor`.